### PR TITLE
Add image to gift model and improve controller

### PR DESCRIPTION
**Pull request by denisbodnar**
- [ ] @yarvanya
- [ ] @yurii-kokhalevych
- [ ] @AlbaMontis

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

### CODE REVIEW

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/controllers/gift.controller.js [hunk 1]
```diff
@@ -7,20 +7,19 @@ const Guest = DB.Guest;
 const { messages } = require('./../helper');
 
 const isEventOwner = (eventId, userId, gift) => {
-  return Event.findOne({where: {id: +eventId, $and: {owner: userId}}})
-  .then(event => !!event && !!gift && (+gift.dataValues.event_id === +eventId))
+  return Event.findOne({where: {id: eventId, $and: {owner: userId}}})
```
Comment (AlbaMontis) [inline server/controllers/gift.controller.js]: You can return without word "return"

File: server/controllers/gift.controller.js [hunk 2]
```diff
@@ -7,20 +7,19 @@ const Guest = DB.Guest;
...
+  .then(event => !!event && !!gift)
   .catch(error => error);
 };
 
 const isEventGuest = (eventId, userId, gift, res) => {
-  return Guest.findOne({where: {event_id: +eventId, $and: {user_id: userId}}})
-  .then(guest => !!guest && !!gift && (+gift.dataValues.event_id === +eventId) ||
-    res.status(404).send(messages.giftNotFound))
+  return Guest.findOne({where: {event_id: eventId, $and: {user_id: userId}}})
```
Comment (AlbaMontis) [inline server/controllers/gift.controller.js]: The same as above.

Comment (AlbaMontis) [reply server/controllers/gift.controller.js]: Also it looks like "$and" is unnecessary.

File: server/controllers/gift.controller.js [hunk 3]
```diff
@@ -7,20 +7,19 @@ const Guest = DB.Guest;
...
+  .then(guest => !!guest && !!gift || res.status(404).send(messages.giftNotFound))
   .catch(error => error);
 };
 
 const eventIsDraft = eventId => {
-  return Event.findOne({where: {id: Number(eventId), $and: {status_event_id: 1}}})
+  return Event.findOne({where: {id: +eventId, $and: {status_event_id: 1}}})
```
Comment (AlbaMontis) [inline server/controllers/gift.controller.js]: The same as above.

Comment (AlbaMontis) [reply server/controllers/gift.controller.js]: `+` is not a good practice. Use parseInt().

File: server/controllers/gift.controller.js [hunk 4]
```diff
@@ -30,7 +29,7 @@ module.exports = {
     let giftParams = Object.assign({}, req.body, {event_id: req.params.id});
```
Comment (AlbaMontis) [inline server/controllers/gift.controller.js:26]: It could be const.

File: server/controllers/gift.controller.js (hunk 1 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: In a concise body, only an expression is needed, and an implicit return is attached. In a block body, you must use an explicit return statement.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions

File: server/controllers/gift.controller.js (hunk 2 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: $and is a Sequelize query operator

http://docs.sequelizejs.com/manual/tutorial/querying.html

File: server/controllers/gift.controller.js (hunk 3 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: this shortcut works for me

File: server/controllers/gift.controller.js (hunk 4 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js:26]: it is just a temporary value, changing with every request, so there is no need for it to be declared with const

**Review by yarvanya — DISMISSED** [review]

File: server/controllers/gift.controller.js (hunk 2 above)
Comment (AlbaMontis) [reply server/controllers/gift.controller.js]: I know. But can't you write just `{where: {event_id: eventId, user_id: userId}}` it is the same but without useless $and operator.

File: server/controllers/gift.controller.js (hunk 1 above)
Comment (AlbaMontis) [reply server/controllers/gift.controller.js]: Why can't you use concise body here?

File: server/controllers/gift.controller.js (hunk 4 above)
Comment (AlbaMontis) [reply server/controllers/gift.controller.js:26]: But you don't reassign it.
https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations

File: server/controllers/gift.controller.js (hunk 3 above)
Comment (AlbaMontis) [reply server/controllers/gift.controller.js]: screenshot from 2017-07-05 11-22-27

File: server/controllers/gift.controller.js (hunk 2 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: Maybe

File: server/controllers/gift.controller.js (hunk 1 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: Yep, good point

File: server/controllers/gift.controller.js (hunk 3 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js]: I know about speed)

File: server/controllers/gift.controller.js (hunk 4 above)
Comment (denisbodnar) [reply server/controllers/gift.controller.js:26]: Ok then

**Review by AlbaMontis — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/controllers/gift.controller.js [hunk 5]
```diff
@@ -47,7 +43,7 @@ module.exports = {
   },
 
   retrieve(req, res) {
-    Gift.findById(req.params.gift_id)
+    Gift.findOne({where: {id: req.params.gift_id, event_id: req.params.id}})
```
Comment (4asof4ik) [inline server/controllers/gift.controller.js]: you have this code in 57 73 create common function this

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by 4asof4ik — APPROVED** [review]
Fix content and merge

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — APPROVED** [review]

**Pull request merged**